### PR TITLE
Update next definitions for 6.1.1 - router events API

### PR DESCRIPTION
**Pull request by jakemmarsh**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/zeit/next.js/pull/4726
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by jthegedus — APPROVED** [review]

**Review by resir014 — CHANGES_REQUESTED** [review]

File: types/next/router.d.ts [hunk 1]
```diff
@@ -48,13 +56,10 @@ export interface RouterProps {
...
     // events
-    onAppUpdated?(nextRoute: string): void;
-    onBeforeHistoryChange?(as: string): void;
-    onHashChangeStart?(url: string): void;
-    onHashChangeComplete?(url: string): void;
-    onRouteChangeComplete?(url: string): void;
-    onRouteChangeError?(error: any, url: string): void;
-    onRouteChangeStart?(url: string): void;
+    readonly events: {
+        on: (eventName: EventName, handler: (url: string) => any) => void;
```
[the remark this answers is not in the document]
Comment (jakemmarsh) [reply types/next/router.d.ts]: Thanks, updated. Not sure if the way I've handled it is ideal so I'm open to feedback.

**Comment by kachkaev** [comment]
`Router.onRouteChangeStart` still works in `next@6.1`, but is removed here. This is a breaking change, which can cause false-positive TypeErrors. Some context: https://github.com/zeit/next.js/issues/4863

Could `onRouteChangeXXX` be brought back?

**Comment by andy-ms** [comment]
@kachkaev PRs accepted.

**Comment by resir014** [comment]
@kachkaev @andy-ms Yeah, apologies guys. I'm working on a PR to get it back as we speak.

**Pull request merged**